lib-driver/caqti_driver_mariadb.ml

```diff
@@ -493,6 +495,39 @@ module Connect_functor (System : Caqti_driver_sig.System_unix) = struct
       (function
        | Ok () -> return (Ok ())
        | Error err -> transaction_failed "# ROLLBACK" err)
+
+    let populate ~table ~columns row_type input_stream =
+      let columns_tuple = "(" ^ (String.concat "," columns) ^ ")" in
+      let values_tuple = "(" ^ (String.concat "," (List.map (fun _ -> "?") columns)) ^ ")" in
+      let insert_query =
+        Caqti_request.exec
+          row_type
+          ("INSERT INTO " ^ table ^  " " ^ columns_tuple ^ " VALUES " ^ values_tuple)
+      in
+      (* TODO: Should we prepare the statement directly somehow? *)
```

Comment:
We could memoize them with a hash table.  This is a potential memory leak, but only combinatorially preparing different table and different subset and ordering of columns pick from a finite number of names known to the application.  Actually, I think memoizing should can done for all prepared queries, implemented by `Caqti_request.create`, so maybe leave it as single-shot queries for now.

Reply:
I think the full solution is probably to separate out the preparation logic in `create`, which I think would speed things up quite a bit, but for now I think leaving it as one-shot is acceptable.